### ISSUE TITLE
pthread not working

### ISSUE THREAD

**Issue by nyuichi** [linux-noah/noah#29]
A code that reproduces the bug:

```c
#include<stdio.h>
#include<string.h>
#include<pthread.h>
#include<stdlib.h>
#include<unistd.h>

pthread_t tid[2];
int counter;

void* doSomeThing(void *arg)
{
    unsigned long i = 0;
    counter += 1;
    printf("\n Job %d started\n", counter);

    for(i=0; i<(0xFFFFFFFF);i++);
    printf("\n Job %d finished\n", counter);

    return NULL;
}

int main(void)
{
    int i = 0;
    int err;

    while(i < 2)
    {
        err = pthread_create(&(tid[i]), NULL, &doSomeThing, NULL);
        if (err != 0)
            printf("\ncan't create thread :[%s]", strerror(err));
        i++;
    }

    pthread_join(tid[0], NULL);
    pthread_join(tid[1], NULL);

    return 0;
}
```

The expected result:

```
Job 1 started
Job 2 started
Job 2 finished
Job 2 finished
```

What I got:

```
 Job 1 started
```

**Comment by nyuichi** [comment]
Another test case:

```c
#include<stdio.h>
#include<string.h>
#include<pthread.h>
#include<stdlib.h>
#include<unistd.h>

void doit(const char *str)
{
  for (int i = 0; i < 0x4; ++i) {
    for (int j = 0; j < 0xfffffff; ++j) {
      puts(str);
    }
  }
}

void* thread(void *arg)
{
  doit("from child");
  return NULL;
}

int main(void)
{
  pthread_t tid;
  pthread_create(&tid, NULL, &thread, NULL);
  doit("from parent");
  //  pthread_join(tid, NULL);
  return 0;
}
```

```c
[yuichi@canopus build]$ ./a.out 
Noah WARNING: SA_SIGINFO implementaion is incomplete: 0x4000004
Noah WARNING: SA_SIGINFO implementaion is incomplete: 0x14000004
Noah WARNING: SA_SIGINFO implementaion is incomplete: 0x4000004
Noah WARNING: SA_SIGINFO implementaion is incomplete: 0x4000004
Noah WARNING: SA_SIGINFO implementaion is incomplete: 0x4000004
Noah WARNING: unknown options given to wait4: 0x8
Noah WARNING: set_robust_list is unimplemented
Noah WARNING: SA_SIGINFO implementaion is incomplete: 0x4000004
Noah WARNING: SA_SIGINFO implementaion is incomplete: 0x14000004
Noah WARNING: set_robust_list is unimplemented
Noah WARNING: futex is unimplemented
from child
Noah WARNING: futex is unimplemented
Noah WARNING: unsupported futex command: 128
from child
from child
from child
from child
from child
!!PANIC!!
perror is "Undefined error: 0" if it is valid
panic
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
[98850:2963520] Obtained 9 stack frames.
from child
0   noah                                0x00000001099cfa55 printbt_to_sink + 85
1   noah                                0x00000001099cf9c0 panic + 608
from child
2   noah                                0x00000001099dd084 sys_futex + 308
3   noah                                0x00000001099dcdcf _sys_futex + 447
4   noah                                0x00000001099c8237 handle_syscall + 199
from child
5   noah                                0x00000001099c7b46 main_loop + 630
6   noah                                0x00000001099c8ace main + 910
from child
7   libdyld.dylib                       0x00007fffe0b9e255 start + 1
8   ???                                 0x0000000000000005 0x0 + 5
from child

from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
from child
[yuichi@canopus build]$ 
```

**Comment by nullpo-head** [comment]
Fixed by  b20a506b72be0ffd9dc806864cf256d6b5d9567d
